File: usbdescreqparser_test.py
from pytest import mark
from usbdescreqparser import Parser


with open("testdata/stddesc-hex.txt", "r", encoding="utf-8") as file:
    stddesc_hex = file.read()

with open("testdata/stddesc-expected.txt", "r", encoding="utf-8") as file:
    stddesc_expected = file.read().rstrip()

def test_best_guess():
    parser = Parser()
    actual = parser.best_guess(stddesc_hex)

    assert actual == stddesc_expected


with open("testdata/microphone-stddesc-hex.txt", "r", encoding="utf-8") as file:
    microphone_stddesc_hex = file.read()

with open("testdata/microphone-stddesc-expected.txt", "r", encoding="utf-8") as file:
    microphone_stddesc_expected = file.read().rstrip()

def test_best_guess_microphone():
    parser = Parser()
    actual = parser.best_guess(microphone_stddesc_hex)

    assert actual == microphone_stddesc_expected

File: usbdescreqparser.test.js
const Parser = require("./usbdescreqparser.js");
const fs = require("fs");

const stddescHex = fs.readFileSync("testdata/stddesc-hex.txt", "utf-8");
const stddescExpected = fs
  .readFileSync("testdata/stddesc-expected.txt", "utf-8")
  .trimEnd();


const microphoneStddescHex = fs.readFileSync("testdata/microphone-stddesc-hex.txt", "utf-8");
const microphoneStddescExpected = fs
  .readFileSync("testdata/microphone-stddesc-expected.txt", "utf-8")
  .trimEnd();


describe("best_guess", () => {
  test("stddesc", () => {
    const parser = new Parser();

    console.log(stddescHex);

    expect(parser.best_guess(stddescHex)).toBe(stddescExpected);
  });

  test("microphone-stddesc", () => {
    const parser = new Parser();

    console.log(microphoneStddescHex);

    expect(parser.best_guess(microphoneStddescHex)).toBe(microphoneStddescExpected);
  });
});
